### PR TITLE
#28562: [Bug]: DataGrid Headers look weird on Safari

### LINKED ISSUE #28562: [Bug]: DataGrid Headers look weird on Safari
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: macOS 12.2.1
    CPU: (4) x64 Intel(R) Core(TM) i7-4578U CPU @ 3.00GHz
    Memory: 71.48 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Browsers:
    Chrome: 114.0.5735.198
    Safari: 15.3
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/xnqzvf?file=/example.tsx

## Actual Behavior
DataGrid Headers look like this on Mac Safari

image

## Expected Behavior
image

### Logs

_No response_

### Requested priority

Low

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by AbdulRehmanMehar**
## Previous Behavior

DataGrid Headers look like this on Mac Safari

image

## New Behavior
DataGrid Headers will look like this on Mac Safari

image

https://github.com/microsoft/fluentui/assets/28218967/e7090f1a-794d-4b4b-8089-aa4f7b59836a

## Related Issue(s)
https://github.com/microsoft/fluentui/issues/28562

Fixes #28562

### CODE REVIEW

**Comment by AbdulRehmanMehar** [comment]
@microsoft-github-policy-service agree

**Comment by AbdulRehmanMehar** [comment]
> @AbdulRehmanMehar please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">74</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount_base.html">10</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Avatar-mount_base.html">580</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Avatar-mount.html">600</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Button-mount_base.html">301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Button-mount.html">292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Field-mount_base.html">1014</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Field-mount.html">1036</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProvider-mount_base.html">655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProvider-mount.html">667</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-mount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">74</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount_base.html">10</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/MakeStyles-mount_base.html">861</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/MakeStyles-mount.html">847</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Persona-mount_base.html">1622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/Persona-mount.html">1627</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/SpinButton-mount_base.html">1261</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28588/perf-test-react-components/SpinButton-mount.html">1342</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                   Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `158.282 kB`<br />`43.276 kB` | `158.225 kB`<br />`43.247 kB` | `-57 B` <br />`-29 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>    |  `44.666 kB`<br />`12.442 kB` |  `44.609 kB`<br />`12.414 kB` | `-57 B` <br />`-28 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         | `132.365 kB`<br />`33.849 kB` | `132.308 kB`<br />`33.819 kB` | `-57 B` <br />`-30 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |  `77.993 kB`<br />`19.245 kB` |  `77.936 kB`<br />`19.218 kB` | `-57 B` <br />`-27 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |  `76.978 kB`<br />`18.973 kB` |  `76.921 kB`<br />`18.947 kB` | `-57 B` <br />`-26 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `65.175 kB`<br />`17.915 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `210.68 kB`<br />`58.781 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `36.409 kB`<br />`12.003 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8ba163a283ca6491ab90345b1c892df48abc5f3b'>8ba163a283ca6491ab90345b1c892df48abc5f3b</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-table-f1b9fee8-bec3-49cb-b249-88f7dbfdcd5b.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "#28562: [Bug]: DataGrid Headers look weird on Safari",
```
Comment (sopranopillow) [inline change/@fluentui-react-table-f1b9fee8-bec3-49cb-b249-88f7dbfdcd5b.json:3]:
```suggestion
  "comment": "fix: DataGrid Headers look weird on Safari",
```

Comment (AbdulRehmanMehar) [reply change/@fluentui-react-table-f1b9fee8-bec3-49cb-b249-88f7dbfdcd5b.json]: Hi @sopranopillow, thanks for the correction. I've updated the comment and pushed changes.

**Comment by sopranopillow** [comment]
@AbdulRehmanMehar, PR ready, just missing an approval from @microsoft/teams-prg. Thanks for your contribution!

**Review by ling1726 — APPROVED** [review]

**Pull request merged**